src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -186,8 +210,143 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
+                              x3(j,i,k)=x3(j,i,k)+qs(j,i,k)
+                           end do
+                        end do
+                     end do
+
+                  case('qg','QG')
+
+                     do k=1,grd_ens%nsig
+                        do i=1,grd_ens%lon2
+                           do j=1,grd_ens%lat2
+                              if (l_use_cvpqx) then
+                                  if (qg(j,i,k) <= 1.0E-5_r_kind) then
+                                     qg(j,i,k) = 1.0E-5_r_kind
+                                  end if
+                                  if (l_use_cvpqx_pval .gt. 0.0_r_kind ) then ! CVpq
+                                     if(mype==0 .and. i==10 .and. j==10 .and. k==10) write(6,*)'power transform for qg : from member-->',n
+                                     qg(j,i,k)=((qg(j,i,k)**l_use_cvpqx_pval)-1)/l_use_cvpqx_pval
+                                  else  ! CVlogq
+                                     if(mype==0 .and. i==10 .and. j==10 .and. k==10) write(6,*)'log transform for qg : from member-->',n
+                                     qg(j,i,k) = log(qg(j,i,k))
+                                  end if
+                              end if
+                              w3(j,i,k) = qg(j,i,k)
+                              x3(j,i,k)=x3(j,i,k)+qg(j,i,k)
+                           end do
+                        end do
+                     end do
+
+                  case('qnr','QNR')
+
+                       do k=1,grd_ens%nsig
+                          do i=1,grd_ens%lon2
+                             do j=1,grd_ens%lat2
+                                if ( l_use_dbz_directDA ) then ! direct reflectivity DA
+                                   if ( cld_nt_updt .gt. 0 ) then ! Update Number Concentration
+                                      if (l_cvpnr) then ! Power Transform
+                                         if (qnr(j,i,k) < one) then
+                                             qnr(j,i,k) = one
+                                         end if 
+                                         if(mype==0 .and. i==10 .and. j==10 .and. k==10) &
+                                         write(6,*)'updating qnr in hybrid analysis and power transform for qnr : from member-->',n
+                                         qnr(j,i,k) =((qnr(j,i,k)**cvpnr_pval)-1)/cvpnr_pval
+                                      else ! No transform
+                                         if(mype==0 .and. i==10 .and. j==10 .and. k==10) &
+                                         write(6,*)'updating qnr in hybrid analysis : NO log-transform to member-->',n
+                                      end if
+                                      w3(j,i,k) = qnr(j,i,k)
+                                      x3(j,i,k)=x3(j,i,k)+qnr(j,i,k)
+                                   else
+                                      write(6,*)'qnr is not analyzed in hybrid analysis : member-->',n
```

Comment:
Check this write: could it write tons of information? Can move this out of loop?

Reply:
I modified the code according to your comments. I took out these codes from loops and make them print one time at the start of the routine using if(mype==0). 

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-86aea35b64364db7913dd1eb935d98f572e0601c925defcb5595748644029235R241-R304